rasa/utils/llm.py

```diff
@@ -74,3 +80,117 @@ def sanitize_message_for_prompt(text: Optional[str]) -> str:
     A string with new lines removed.
     """
     return text.replace("\n", " ") if text else ""
+
+
+def combine_custom_and_default_config(
+    custom_config: Optional[Dict[Text, Any]], default_config: Dict[Text, Any]
+) -> Dict[Text, Any]:
+    """Merges the given llm config with the default config.
+
+    Only uses the default configuration arguments, if the type set in the
+    custom config matches the type in the default config. Otherwise, only
+    the custom config is used.
+
+    Args:
+        custom_config: The custom config containing values to overwrite defaults
+        default_config: The default config.
+
+    Returns:
+        The merged config.
+    """
+    if custom_config is None:
+        return default_config
+
+    if "type" in custom_config:
+        # rename type to _type as "type" is the convention we use
+        # across the different components in config files.
+        # langchain expects "_type" as the key though
+        custom_config["_type"] = custom_config.pop("type")
+
+    if "_type" in custom_config and custom_config["_type"] != default_config.get(
+        "_type"
+    ):
+        return custom_config
```

Comment:
it's surprising that a different `_type` leads to all default parameters being tossed, but I guess also makes sense, many llm providers have different defaults. making a comment about this in the docs too